code/game/turfs/closed/walls.dm

```diff
@@ -37,6 +37,16 @@
 	if(is_station_level(z))
 		GLOB.station_turfs += src
 
+	//integrity stuff
```

Comment:
This is copypasted from atom initialize. I think walls are deployed before initializing an atom works properly, so if this isn't here, walls aren't initialized with armor/atom_integrity being set to max_integrity. This is a hack and I'd like if there was a cleaner way to do this.

Reply:
turf/Initialize doesn't call parent. That's why you are seeing this behaviour.

Reply:
Can you at least encapsulate this entire block in another proc and then call that from both turf initialize and atom initialize? Also this currently results in dense turfs that don't inherit from /wall to have 0 integrity.

Reply:
Thanks, this helped